### PR TITLE
Show the charging state on the Battery Level sensor's icon

### PR DESCRIPTION
**Pull request by marazmarci**
## Summary
Show the charging state on the Battery Level sensor's icon.
A different icon will be shown for wired and wireless charging.
A new DB property called `last_sent_icon` is added, so I needed to implement a new DB migration.

A side effect of these changes:
From now on, changing only the icon (but not the state) of a sensor will also trigger sending an update to the server.

## Screenshots (frontend)

Wired charging:

home-assistant-android-battery-charging-0

home-assistant-android-battery-charging-1

home-assistant-android-battery-charging

home-assistant-android-battery-charging-100

Wireless charging:

home-assistant-android-battery-wireless-charging-0

home-assistant-android-battery-wireless-charging-1

home-assistant-android-battery-wireless-charging-9

home-assistant-android-battery-wireless-charging-10

home-assistant-android-battery-wireless-charging-100

Not charging:

home-assistant-android-battery-not_charging-9

home-assistant-android-battery-not_charging-10

home-assistant-android-battery-not_charging

home-assistant-android-battery-not_charging-100

Wear OS:

home-assistant-android-wear-os-battery-charging

### CODE REVIEW

File: common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt [hunk 1]
```diff
@@ -284,7 +284,8 @@ abstract class SensorReceiverBase : BroadcastReceiver() {
             try {
                 success = integrationUseCase.updateSensors(enabledRegistrations.toTypedArray())
                 enabledRegistrations.forEach {
-                    sensorDao.updateLastSendState(it.uniqueId, it.state.toString())
+                    sensorDao.updateLastSentState(it.uniqueId, it.state.toString())
+                    sensorDao.updateLastSentIcon(it.uniqueId, it.icon)
```
Comment (jpelgrom) [inline common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt:288]: This should be done in a transaction to improve performance (wrap them in [another function with `@Transaction`](https://developer.android.com/reference/kotlin/androidx/room/Transaction))

Comment (marazmarci) [reply common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt:288]: Oh, I just saw this part in your message:

> Edit: or using the regular update function and just change two properties, like failed registrations

OK, then I'll go with the single update method with two properties approach.

Comment (marazmarci) [reply common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorReceiverBase.kt]: Done in a25c396

**Review by dshokouhi — APPROVED** [review]

**Comment by marazmarci** [comment]
I didn't change anything, just had an idea for an improvement, so I quickly converted the PR to a draft.
Then I realized that my idea is not possible to implement, so it's ready to be merged again.

**Review by JBassett — APPROVED** [review]

**Pull request merged**